### PR TITLE
drivers/sx127x: skip reset sequence if pin is undefined

### LINKED ISSUE #11913: drivers/sx127x: Enforces reset pin to be defined
#### Description
After #11678 the reset sequence in the sx127x devices has been fixed, but having a reset pin is now mandatory. This is not the case of the `sensebox_samd21` board, which generates a reset pulse by adding a delay to the power line (see [schematics](https://github.com/watterott/senseBox-MCU/blob/master/hardware/senseBox-MCU_v12.pdf)). This currently makes the driver initialization fail on that board. How should we handle this?

I propose to remove the check of the reset pin being undefined from the driver, and only apply the reset sequence if the pin if defined. The side effect of that is that `sx127x_reset` will not actually reset the module in the `sensebox_samd21`. Although, the radio will reset on a software reset of the MCU. If we agree on that, I can provide the fix on a PR.

#### Steps to reproduce the issue
Build and flash `tests/driver_sx127x` application on a `sensebox_samd21`, or set `SX127X_PARAM_RESET` to `GPIO_UNDEF`.

#### Expected results
Successful initialization, output with debug on:
```
2019-07-25 12:16:04,856 - INFO # main(): This is RIOT! (Version: 2019.10-devel-111-g7eb57)
2019-07-25 12:16:04,860 - INFO # [sx127x] netdev: initializing driver...
2019-07-25 12:16:04,863 - INFO # [sx127x] SPI_1 initialized with success
2019-07-25 12:16:04,895 - INFO # [sx127x] initializing radio settings
2019-07-25 12:16:04,900 - INFO # [sx127x] netdev: initialization done
2019-07-25 12:16:04,905 - INFO # Initialization successful - starting the shell now
```

#### Actual results
Driver fails to initialize:
```
2019-07-25 12:57:28,527 - INFO # main(): This is RIOT! (Version: 2019.10-devel-111-g7eb57)
2019-07-25 12:57:28,528 - INFO # [sx127x] netdev: initializing driver...
2019-07-25 12:57:28,530 - INFO # [sx127x] SPI_0 initialized with success
2019-07-25 12:57:28,531 - INFO # [sx127x] error: No reset pin defined.
2019-07-25 12:57:28,533 - INFO # [sx127x] netdev: initialization failed
2019-07-25 12:57:28,534 - INFO # Failed to initialize SX127x device, exiting
```

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR is an attempt to fix #11913: the sx127x driver reset sequence fails if the reset pin is undefined but some board don't provide it.

As initially proposed in #11913, it just don't return an error if the reset pin is not undefined but just skip the reset sequence.

Note that I moved the block initializing TX/RX switch pins before the reset sequence. I'm not sure what consequence this could have.

### Testing procedure

- Reset sequence should still work for board providing the reset pin
- The driver should work for not providing the reset pin

### Issues/PRs references

fixes #11913

### CODE REVIEW

**Comment by aabadie** [comment]
Note: tested this PR with LoRaWAN on TTN with b-l072z-lrwan1 (sx1276) and it works as before. It would be great to also test with sx1272.

**Comment by leandrolanzieri** [comment]
`sx127x_init` still returns an error if the reset pin is not defined.

**Comment by aabadie** [comment]
> sx127x_init still returns an error if the reset pin is not defined.

Indeed, I forgot this one. Removed and directly amended.

File: drivers/sx127x/sx127x.c [hunk 1]
```diff
@@ -140,12 +135,6 @@ int sx127x_init(sx127x_t *dev)
...
 
-    /* Check if the reset pin is defined */
-    if (dev->params.reset_pin == GPIO_UNDEF) {
-        DEBUG("[sx127x] error: No reset pin defined.\n");
-        return -SX127X_ERR_GPIOS;
-    }
-
     _init_timers(dev);
 
     /* reset pin should be left floating during POR */
```
Comment (leandrolanzieri) [inline drivers/sx127x/sx127x.c]: Reset pin is still being initialized here.

Comment (aabadie) [reply drivers/sx127x/sx127x.c]: Ah yes, should be fixed now.

**Comment by aabadie** [comment]
Thank you for your review!

**Pull request merged**